### DISCUSSION TITLE
`select` option in `ZkNotes` seems weird

### DISCUSSION THREAD

**Discussion by danymat** [zk-org/zk-nvim#142] [General]
Hello, 
I broke my head this morning about using the `select` option in `ZkNotes`, referring to [zklist lsp](https://github.com/mickael-menu/zk/blob/main/docs/editors-integration.md#zklist).
I wanted my picker to only display the file name, so after trying every `select` cases, i found that:

```lua
 require("zk.commands").get("ZkNotes")({ select = {'path' } })
```

Was the way to go:

![Capture d’écran 2023-08-11 à 15 11 59](https://github.com/mickael-menu/zk-nvim/assets/5306901/eb29980d-510a-4871-a6f7-cb444910c8f0)

But if we don't add:

```lua
 require("zk.commands").get("ZkNotes")({ select = {'path' , 'absPath'} })
```

We get a nil error when opening the fuzzed file (`/zk/lua/zk.lua:135: attempt to concatenate field 'absPath' (a nil value)`)

I think these parameters are somehow difficult to understand in the zklist lsp documentation (or maybe there indeed a bug)

**Comment by mickael-menu** [comment]
As the output of this command might be very verbose and put a heavy load on the LSP client, you need to explicitly set which note fields you want to receive when using `zk.list` with the `select` option.

This fails because `zk-nvim` requires the `absPath` field to work properly when using the `ZkNotes` command.

I don't think you can customize this, because it's hard-coded in the `zk-nvim` fzf picker here:

https://github.com/mickael-menu/zk-nvim/blob/797618aa07f58ceba6f79fb6e777e8e45c51e1ce/lua/zk/pickers/fzf.lua#L19

Although if you use what you suggested (removing `title` with `{ select = {'path' , 'absPath'} }`) this should work, but is a side-effect more than a feature.

**Reply by lyndhurst** [reply]
I have been trying to set the select option as well with no success. Actually, `{ select = {'path' , 'absPath'} }` is the only one that yielded a result list and I did not even come up with it myself !

I feel dumb, but I do not get your answer completely. What is the bottom line here ?
Is it possible at all to customize the result line with `zk-nvim`, or should I try and use `zk list` in my own `fzf-lua` command for example ?

**Reply by mickael-menu** [reply]
It's not possible to customize the result line with `zk-nvim`, `select` is not made for that but only to specify which payload we get from the LSP server.

**Reply by lyndhurst** [reply]
Thank you for clarifying it for me. I gave it a go with `fzf-lua`, it is unfortunately beyond my very limited `lua` skills.
